crypto/trace.c

```diff
@@ -146,19 +146,32 @@ const char *OSSL_trace_get_category_name(int num)
 {
     size_t i;
 
+    /*
+     * Partial check that OSSL_TRACE_CATEGORY_... macros
+     * are synced with trace_categories array
+     */
+    static const size_t trace_categories_len = OSSL_NELEM(trace_categories);
+    if (!ossl_assert(OSSL_TRACE_CATEGORY_NUM == trace_categories_len))
+        return NULL;
+
     for (i = 0; i < OSSL_NELEM(trace_categories); i++)
```

Comment:
Please use `trace_categories_len` here.

Reply:
The other option is to drop `trace_categories_len` and replace `OSSL_TRACE_CATEGORY_NUM == trace_categories_len` with `OSSL_TRACE_CATEGORY_NUM == OSSL_NELEM(trace_categories)` above.